### DISCUSSION TITLE
HA si riavvia da solo , 3-4 volte al giorno su RIPI4

### DISCUSSION THREAD

**Discussion by lucaxxaa** [indomus/forum#1029] [HUB: Home Assistant]
Da circa una settimana, HA, si riavvia da solo. E' su RPI4  con boot su sd e poi HD esterno con trim, e alimentatore dedicato da 4 A

Non c'e' consumo esagerato di ram e cpu, e da circa una settimana ho aggiungo vari shelly 2.5, nient altro.

Come posso capire da vari log, o abilitare qualche automazione o logger, per capire cosa lo fa riavviare ? 

GRAZIE

**Comment by snakuzzo** [comment]
Gli Shelly non c'entrano niente con i riavvii di Home Assistant.
Non devi attivare alcuna automazione...se HA si riavvia dovresti già vedere nei log cosa accade.
Se sei su RaspOS ti consiglio anche di dare un'occhiata al syslog

**Comment by lucaxxaa** [comment]
Sono su hassio. 
Che file di log dovrei controllare ?
Se fosse ad esempio la alimentatore , come posso capire il malfunzionamento?
Grazie.

**Reply by snakuzzo** [reply]
non creare nuovi thread...devi fare reply sotto l'ultima risposta!

per HA il file di log di HA...lo puoi vedere pure da GUI.
Per quanto riguarda l'alimentazione del RPI, guarda [QUI](https://www.home-assistant.io/integrations/rpi_power/)

**Reply by lucaxxaa** [reply]
Ok ti ringrazio per l'alimentazione avevo già guardato e sembra tutto lineare, proverò a dare un occhio ai file di log anche se non ci capisco un granché

**Reply by snakuzzo** [reply]
Considera che "imparare a leggere il log" è forse la cosa più importante. Senza saper guardare nei log vuol dire andare a caso affidandosi alla fortuna...e non mi pare una cosa tanto buona :)

**Comment by lucaxxaa** [comment]
In relazione ai continui riavvi autonomi di hassios, su RPI4, ho  provato ad aumentare lo spazio di memoria Swap a 2 Gb.

Per ora sono due giorni, senza nessun riavvio spontaneo

Ho seguito, questo metodo, un po' alternativo 
https://community.home-assistant.io/t/how-to-increase-the-swap-file-size-on-home-assistant-os/272226
